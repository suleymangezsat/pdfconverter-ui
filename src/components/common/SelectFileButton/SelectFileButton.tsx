import { Button } from "@material-ui/core";
import React, { ChangeEvent, memo } from "react";

type Props = {
  children: React.ReactNode;
  fileType?: string;
  onSelect: (files: File[] | null) => void;
};

export const SelectFileButton = memo(
  ({ children, fileType, onSelect }: Props) => (
    <label>
      <input
        name="btn-upload"
        style={{ display: "none" }}
        type="file"
        accept={fileType}
        multiple={true}
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          onSelect(e.target.files ? Array.from(e.target.files) : null)
        }
      />
      <Button variant="outlined" component="span">
        {children}
      </Button>
    </label>
  )
);
