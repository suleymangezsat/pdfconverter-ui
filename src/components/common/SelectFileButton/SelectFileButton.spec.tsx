import React from "react";
import { shallow } from "enzyme";
import { SelectFileButton } from "./SelectFileButton";

describe("components/common/SelectFileButton", () => {
  const handleSelect = jest.fn();
  const fileType = "application/pdf";
  const children = <p>Test Text</p>;

  const prepareComponent = (props?: any) => (
    <SelectFileButton fileType={fileType} onSelect={handleSelect} {...props}>
      {children}
    </SelectFileButton>
  );

  it("should render SelectFileButton component correctly", () => {
    const component = shallow(prepareComponent());
    expect(component).toMatchSnapshot();
  });

  it("should render children when passed in", () => {
    const component = shallow(prepareComponent());
    expect(component.contains(children)).toEqual(true);
  });

  it("should set input file type correctly", () => {
    const component = shallow(prepareComponent());
    expect(component.find("input").getElement().props.accept).toEqual(fileType);
  });

  it("should call onSelect function when input changes", () => {
    const component = shallow(prepareComponent());
    const file = new File([""], "filename1.pdf", {
      type: "application/pdf",
      lastModified: new Date().getDate(),
    });
    component.find("input").simulate("change", { target: { files: [file] } });
    expect(handleSelect).toHaveBeenCalledWith([file]);
  });
});
