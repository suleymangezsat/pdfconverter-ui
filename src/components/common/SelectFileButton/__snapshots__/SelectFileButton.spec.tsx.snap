// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/SelectFileButton should render SelectFileButton component correctly 1`] = `
<label>
  <input
    accept="application/pdf"
    multiple={true}
    name="btn-upload"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <WithStyles(ForwardRef(Button))
    component="span"
    variant="outlined"
  >
    <p>
      Test Text
    </p>
  </WithStyles(ForwardRef(Button))>
</label>
`;
